code/client/munkilib/FileRepo.py

```diff
@@ -0,0 +1,171 @@
+        and is a directory.'''
+        return os.path.isdir(os.path.join(self.path, path))
+
+    def isfile(self, path):
+        '''Returns true if the specified path exists in the repo
+        and is a regular file.'''
+        return os.path.isfile(os.path.join(self.path, path))
+
+    def join(self, *args):
+        '''Combines path elements within the repo.'''
+        return os.path.join(*args)
+
+    def dirname(self, path):
+        '''Returns the directory portion of a path.'''
+        return os.path.dirname(path)
+
+    def basename(self, path):
+        '''Returns the filename portion of a path.'''
+        return os.path.basename(path)
+
+    def splitext(self, path):
+        '''Splits the base and extention parts of a path.'''
+        return os.path.splitext(path)
+
+    def mkdir(self, path, mode=0777):
+        '''Creates a directory within the repo.'''
+        return os.mkdir(os.path.join(self.path, path), mode)
+
+    def makedirs(self, path, mode=0777):
+        '''Creates a directory within the repo, including parent directories.'''
+        return os.makedirs(os.path.join(self.path, path), mode)
+
+    def listdir(self, path):
+        '''Lists the contents of a repo directory.'''
+        return listdir(os.path.join(self.path, path))
+
+    def remove(self, path):
+        '''Removes a file from the repo.'''
+        return os.remove(os.path.join(self.path, path))
+
+    def unlink(self, path):
+        '''Removes a file from the repo.'''
+        return os.unlink(os.path.join(self.path, path))
+
+    def get(self, src, dest):
+        '''Copies a file from the repo to a local file.'''
+        cmd = ['/bin/cp', os.path.join(self.path, src), dest]
+        return subprocess.call(cmd)
+
+    def put(self, src, dest):
+        '''Copies a local file to the repo.'''
+        cmd = ['/bin/cp', src, os.path.join(self.path, dest)]
+        return subprocess.call(cmd)
+
+    #
+    # Some callers open a file, but then use the local_path field
+    # to access it rather than reading or writing through the returned
+    # handle.  For local repos those callers could just use the
+    # file name directly rather than opening it through this method,
+    # but for the CommandRepo implementation the local_path field
+    # will be a local temporary file that was copied from the remote
+    # repo and/or will be copied to the remote repo on close.
+    #
+    def open(self, path, mode='r'):
+        '''Opens a file in the repo.'''
+        class RepoFile:
+            def __init__(self, repo, repo_path, mode):
+                self.repo = repo
+                self.repo_path = repo_path
+                self.repo_mode = mode
+                self.file = open(self.repo_path, mode)
+                self.local_path = self.repo_path
+
+            def read(self):
+                return self.file.read()
+
+        return RepoFile(self, os.path.join(self.path, path), mode)
+
+    def mount(self):
```

Comment:
This function has completely removed the functionality added to allow fileshares to be mounted under /Volumes in 10.12. This was added here: https://github.com/munki/munki/commit/e9bdb65ade01294aa702ebbf40a3e050bd0a244b and here: https://github.com/munki/munki/commit/70135813715cc3bd262fd036352fb5c680720c0c